src/scripts/Build-Bicep.ps1

```diff
@@ -1,200 +1,227 @@
+        elseif ($line -match $genericScopeString)
+        {
+            $stopIndex = Get-NextEndRegionIndex -Content $lines -StartingIndex $i
+            $i = $stopIndex + 1
         }
-
-        # Write supporting files, if available
-        if ($CopySupportingFiles -and -not $Debug) {
-            @('main.json', 'metadata.json', 'README.md', 'version.json') `
-            | ForEach-Object { 
-                $sourceFile = Join-Path $Module $_
-                if (Test-Path $sourceFile) {
-                    Copy-Item $sourceFile (Join-Path $outdir $moduleName)
-                }
-            }
+        elseif ($line -match $closingComment)
+        {
+            $i++
+        }
+        else
+        {
+            $null = $outputString.AppendLine($line)
+            $i++
         }
     }
+
+    $outputString.ToString() | Out-File -FilePath $outputFileName
 }
 
-# Generate module and test code
-Build-Modules main.bicep -CopySupportingFiles
-Build-Modules (Join-Path test main.test.bicep)
-
-# Copy tests to README.md examples
-if (-not $Debug) {    
-    Get-ChildItem (Join-Path $outdir * test main.test.bicep) `
-    | Where-Object { $_.Directory.Parent.Name -match "-$($dir.Name)$" } `
-    | ForEach-Object {
-        $testFile = $_
-        $sb = [System.Text.StringBuilder]::new()
-        $writingModule = $false
-        
-        Get-Content $testFile `
-        | ForEach-Object {
-            $line = $_
-            # If test comment, write example header
-            if ($line -match '^\s*//\s*Test\s') {
-                # Parse test number and description
-                $regex = [regex]::Matches($line.Trim(), '^//\s*Test\s*([0-9]+)\s*-\s*(.*)');
-                $number = $regex.Groups[1].Value
-                $text = $regex.Groups[2].Value
-    
-                [void]$sb.AppendLine().AppendLine("### Example $number").AppendLine()
-                [void]$sb.AppendLine($text).AppendLine()
-                [void]$sb.AppendLine('```bicep')
-            }
-            # If module, adjust the target destination
-            elseif ($line -match '^module\s*' -and $line -match '../main.bicep') {
-                $line = $line -replace "../main.bicep", "br/public:cost/$($testFile.Directory.Parent.Name):1.0"
-                [void]$sb.AppendLine($line)
-                $writingModule = $true
-            }
-            # If module body, append code
-            elseif ($writingModule -and -not ($line -match '^}$')) {
-                [void]$sb.AppendLine($line)
-            }
-            # If end of module, close code block
-            elseif ($writingModule -and $line -eq '}') {
-                [void]$sb.AppendLine($line).AppendLine('```')
-                $writingModule = $false
-            }
+# Copy supporting files.
+if ($CopySupportingFiles)
+{
+    Get-ChildItem -Path $parentFolderPath -Exclude (Split-Path -Path $Module -Leaf) -Recurse | Copy-Item -Destination $outputDirectory
+}
+
+if ($IncludeTests)
```

Comment:
I agree, I didn't even see the tests portion of the script until yesterday afternoon so I just copied a lot of the logic since it was the same. It will need to be cleaned up